tests/integration/conftest.py

```diff
@@ -0,0 +1,12 @@
+import os
+
+import pytest
+
+DIR_PATH = os.path.dirname(__file__)
+
+
+@pytest.fixture()
+def flannel_resource():
+    version = "amd64"
+    path = os.path.join(DIR_PATH, "../../flannel-amd64.tar.gz")
+    return f"flannel-{version}={path}"  # noqa: E999
```

Comment:
Oh, no, that's the opposite of what I meant!  :sweat_smile: I meant dropping the hard-coded `"../../flannel-amd64.tar.gz"` and forcing the option to be passed in. That way the user doesn't get a confusing error message about some random file not being found if they don't know to include it when running the test, and instead get a message like `missing required parameter --flannel-resource`.

Maybe you could just explicitly check that the hard-coded file exists and, if not, raise an error indicating that they need to build it and put it in that specific location with that specific name.

Reply:
Oh sorry, I misunderstood you.
What do you say to this?
```bash
_________________________________________________ ERROR at setup of test_build_and_deploy __________________________________________________
Traceback (most recent call last):
  File "/home/ubuntu/charm-flannel/tests/integration/conftest.py", line 25, in flannel_resource
    raise FileNotFoundError(path)
FileNotFoundError: /home/ubuntu/charm-flannel/tests/integration/../../flannel-amd64.tar.gz
```
FYI, I tried to run integration tests twice on my VM and it went smoothly, so I think it's very on edge to run on GitHub runners.
FYI, I modified the README.md section on building the charm.